### PR TITLE
Fix the README of the X packages

### LINKED ISSUE #4577: [pickers] `@mui/x-date-pickers-pro` have the `README.md` from `@mui/x-date-pickers`
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

image

### Expected behavior 🤔

It would be great to have up to date readme and mention requirements of license for this package. 

### Steps to reproduce 🕹

Steps:

1. Go to [npm link](https://www.npmjs.com/package/@mui/x-date-pickers-pro)

### Context 🔦

_No response_

### Your environment 🌎

_No response_

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #4577 

- [x] Fix installation example for `@mui/x-date-pickers-pro`
- [x] Improve description of `@mui/x-date-pickers` and `@mui/x-date-pickers-pro`
- [x] Update the peer dependencies to match the value in `package.json` of every package

### CODE REVIEW

File: packages/x-date-pickers-pro/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # @mui/x-date-pickers
```
Comment (cherniavskii) [inline packages/x-date-pickers-pro/README.md:1]:
```suggestion
# @mui/x-date-pickers-pro
```

File: packages/grid/x-data-grid/README.md [hunk 2]
```diff
@@ -19,8 +19,9 @@ This component has three peer dependencies that you will need to install as well
 
 ```json
 "peerDependencies": {
-  "@mui/material": "^5.0.0",
-  "react": "^17.0.0"
+  "@mui/material": "^5.2.8",
+  "@mui/system": "^5.2.8",
+  "react": "^17.0.2 || ^18.0.0"
```
Comment (cherniavskii) [inline packages/grid/x-data-grid/README.md:24]:
```suggestion
  "react": "^17.0.2 || ^18.0.0",
  "react-dom": "^17.0.2 || ^18.0.0"
```

File: packages/grid/x-data-grid-pro/README.md [hunk 3]
```diff
@@ -19,8 +19,9 @@ This component has three peer dependencies that you will need to install as well
 
 ```json
 "peerDependencies": {
-  "@mui/material": "^5.0.0",
-  "react": "^17.0.0"
+  "@mui/material": "^5.2.8",
+  "@mui/system": "^5.2.8",
+  "react": "^17.0.2 || ^18.0.0"
```
Comment (cherniavskii) [inline packages/grid/x-data-grid-pro/README.md:24]:
```suggestion
  "react": "^17.0.2 || ^18.0.0",
  "react-dom": "^17.0.2 || ^18.0.0"
```

File: packages/x-date-pickers/README.md [hunk 4]
```diff
@@ -15,12 +15,14 @@ npm install @mui/x-date-pickers
...
 ```json
 "peerDependencies": {
-  "@mui/material": "^5.0.0",
-  "react": "^17.0.0"
+  "@mui/material": "^5.2.3",
+  "@mui/system": "^5.2.3",
+  "react": "^17.0.2 || ^18.0.0",
+  "react-dom": "^17.0.2 || ^18.0.0"
 },
 ```
```
Comment (cherniavskii) [inline packages/x-date-pickers/README.md:28]: Change link to documentation below to https://mui.com/x/react-date-pickers/getting-started/ ?
Same with `@mui/x-date-pickers-pro` readme

File: packages/x-date-pickers-pro/README.md [hunk 5]
```diff
@@ -9,18 +9,20 @@ Install the package in your project directory with:
...
-npm install @mui/x-date-pickers
+npm install @mui/x-date-pickers-pro
 
 // with yarn
-yarn add @mui/x-date-pickers
+yarn add @mui/x-date-pickers-pro
 ```
 
-This component has three peer dependencies that you will need to install as well.
+This component has four peer dependencies that you will need to install as well.
```
Comment (cherniavskii) [inline packages/x-date-pickers-pro/README.md:18]: Maybe ditch the number here?
```suggestion
This component has the following peer dependencies that you will need to install as well:
```

**Review by cherniavskii — APPROVED** [review]

File: packages/x-date-pickers-pro/README.md (hunk 5 above)
Comment (flaviendelangle) [reply packages/x-date-pickers-pro/README.md:18]: I think we can yes, @m4theushw

**Review by m4theushw — APPROVED** [review]

**Pull request merged**